src/aks-preview/azext_aks_preview/managed_cluster_decorator.py

```diff
@@ -3104,6 +3155,23 @@ def set_up_azure_monitor_profile(self, mc: ManagedCluster) -> ManagedCluster:
                 metric_labels_allowlist=str(ksm_metric_labels_allow_list),
                 metric_annotations_allow_list=str(ksm_metric_annotations_allow_list))
             self.context.set_intermediate("azuremonitormetrics_addon_enabled", True, overwrite_exists=True)
+
+        if self.context.get_enable_azure_monitor_app_monitoring():
+            if mc.azure_monitor_profile is None:
+                mc.azure_monitor_profile = self.models.ManagedClusterAzureMonitorProfile()
+            mc.azure_monitor_profile.app_monitoring = (
+                self.models.ManagedClusterAzureMonitorProfileAppMonitoring()
+            )
+            mc.azure_monitor_profile.app_monitoring.auto_instrumentation = (
```

Comment:
can customer specify these options as part of the onboarding?

Reply:
Yes they can specify this